src/fsharp/NameResolution.fsi

```diff
@@ -541,11 +551,14 @@ val internal ResolveTypeLongIdentInTyconRef         : TcResultsSink -> NameResol
 /// Resolve a long identifier to a type definition
 val internal ResolveTypeLongIdent                   : TcResultsSink -> NameResolver -> ItemOccurence -> FullyQualifiedFlag -> NameResolutionEnv -> AccessorDomain -> Ident list -> TypeNameResolutionStaticArgsInfo -> PermitDirectReferenceToGeneratedType -> ResultOrException<TyconRef>
 
+/// Resolve a long identifier to a type definition with enclosing type instantiations.
+val internal ResolveTypeLongIdentAndEnclosingTypeInst : TcResultsSink -> NameResolver -> ItemOccurence -> FullyQualifiedFlag -> NameResolutionEnv -> AccessorDomain -> Ident list -> TypeNameResolutionStaticArgsInfo -> PermitDirectReferenceToGeneratedType -> ResultOrException<TypeInst * TyconRef>
```

Comment:
We add a bit more complexity by having separate two different `ResolveTypeLongIdent` with one returning enclosing type instantations. Maybe a better way to unify them?